File: Decorator/PHP/Mocha.php
<?php
  class Mocha extends CondimentDecorator {
    public $beverage;

    public function __construct($beverage) {
      $this->description = ' Mocha';
      $this->beverage = $beverage;
    }

    public function cost() {
      return $this->beverage->cost() + 10.3;
    }

    public function getDescription() {
      return $this->beverage->getDescription() . $this->description;
    }
  }

File: Decorator/Go/decorator/mocha.go
package decorator

type Mocha struct {
  *Condiment
  price float64
  description string
}

func NewMocha(beverage beverage) *Mocha {
  return &Mocha{&Condiment{beverage: beverage}, 0.2, "Mocha"}
}

func (self *Mocha) Cost() float64 {
  return self.beverage.Cost() + self.price
}

func (self *Mocha) GetDescription() string {
  return self.beverage.GetDescription() + " with " + self.description
}
